### DISCUSSION TITLE
How to explicitly set dependency on targets generated by `tar_map()`

### DISCUSSION THREAD

**Discussion by psychelzh** [ropensci/tarchetypes#43] [Help] [answered]
For example:

```r
targets::tar_dir({
    targets::tar_script({
      library(tarchetypes)
      targets <- tar_map(
        values = data.frame(x = 1:2),
        names = NULL,
        tar_target(not_required, x),
        tar_target(object, x)
      )
      list(
        targets,
        tar_target(dummy, targets[[2]])
      )
    })
    targets::tar_visnetwork()
  })
```

I want that the `dummy` target depends on `object_1` and `object_2`, which is essentially `targets[[2]]`. Unfortunately, the code above does not work:

![Rplot](https://user-images.githubusercontent.com/13306020/116496071-c5cdd200-a8d6-11eb-9700-3d470c68e0b4.png)

The expected is:

![Rplot01](https://user-images.githubusercontent.com/13306020/116496195-12191200-a8d7-11eb-96f1-bbb19b12c016.png)

**Comment by wlandau** [comment, marked as answer]
I recommend `tar_combine()` for this. The sketch below assumes `unlist = FALSE` in `tar_map()` (default).

``` r
library(targets)
tar_script({
  library(targets)
  library(tarchetypes)
  targets <- tar_map(
    values = data.frame(x = 1:2),
    names = NULL,
    tar_target(not_required, x),
    tar_target(object, x)
  )
  list(
    targets,
    tar_combine(dummy, targets[[2]])
  )
})

tar_glimpse()
```

![](https://i.imgur.com/u2wXuj9.png)

<sup>Created on 2021-04-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Reply by psychelzh** [reply]
Got it. This is useful for my `tar_render()`. So I should combine them first, then set the combined one as dependent object.

**Reply by wlandau** [reply]
Right, exactly.

**Reply by psychelzh** [reply]
Unfortunately, if the targets are files, the combined is just a character vector, which can not detect the changes if the file names are not changed but the content of each file has changed.

**Reply by wlandau** [reply]
In that case, you can specify `format = "file"` for any of the targets you need to track as files, including the downstream combined one.

**Reply by psychelzh** [reply]
Thanks, that makes sense!
